.pre-commit-config.yaml

```diff
@@ -64,6 +64,30 @@ repos:
     hooks:
     -   id: pylint
         stages: [manual]
+-   repo: https://github.com/pycqa/pylint
+    rev: v2.15.5
+    hooks:
+    -   id: pylint
+        files: ^pandas/
+        exclude: |
+            (?x)
+            ^pandas/tests  # leave turned off
+            |/_testing/  # leave turned off
+            |^pandas/util/_test_decorators\.py  # leave turned off
+            |^pandas/_version\.py  # leave turned off
+            |^pandas/conftest\.py
+            |^pandas/core/generic\.py
+            |^pandas/core/internals/concat\.py
+            |^pandas/core/reshape/merge\.py
+            |^pandas/core/tools/datetimes\.py
+            |^pandas/formats/format\.py
+            |^pandas/io/formats/format\.py
+            |^pandas/io/formats/style\.py
+            |^pandas/io/json/_json\.py
+            |^pandas/io/xml\.py
+            |^pandas/util/_decorators\.py
+            |^pandas/util/_doctools\.py
+        args: [--disable=all, --enable=redefined-outer-name]
```

Comment:
also while I'm touching pylint stuff - adding in a hook only runs the `redefined-outer-name` hook, which is the pylint check I care the most about

the `exclude` list will be gradually trimmed down as contributors tackle https://github.com/pandas-dev/pandas/issues/49656

Reply:
17 seconds still seems very slow for when committing .. Although I assume that is for the full codebase?

Reply:
17 seconds when running `--all-files`, per file it should be neglibile - running on just `pandas/core/generic.py` takes ~.01 seconds